File: tools/injection/templates/constructor.c
#include <stdio.h>
#include <stdlib.h>

// Constructor function - executed when library is loaded
__attribute__((constructor))
void injected_constructor() {
    printf("[INJECTED] Constructor executed!\n");
    // Add your injection code here
}

// Destructor function - executed when library is unloaded
__attribute__((destructor))
void injected_destructor() {
    printf("[INJECTED] Destructor executed!\n");
    // Add cleanup code here
}

// Example function that can be called from injected code
void injected_function() {
    printf("[INJECTED] Custom function called!\n");
}

File: tools/injection/templates/constructor.f90
module injection_module
    use iso_c_binding
    implicit none
    
contains
    
    ! Constructor-like subroutine
    subroutine injected_constructor() bind(c, name="injected_constructor")
        write(*,*) '[INJECTED] Fortran constructor executed!'
        ! Add your injection code here
    end subroutine injected_constructor
    
    ! Example function
    subroutine injected_function() bind(c, name="injected_function")
        write(*,*) '[INJECTED] Fortran function called!'
    end subroutine injected_function
    
end module injection_module
